src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -274,13 +275,36 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const OperationalAdvertisingParamete
 
     ReturnErrorOnFailure(MakeInstanceName(service.mName, sizeof(service.mName), params.GetFabricId(), params.GetNodeId()));
     strncpy(service.mType, "_chip", sizeof(service.mType));
-    service.mProtocol      = MdnsServiceProtocol::kMdnsProtocolTcp;
-    service.mPort          = CHIP_PORT;
-    service.mTextEntries   = nullptr;
-    service.mTextEntrySize = 0;
-    service.mInterface     = INET_NULL_INTERFACEID;
-    service.mAddressType   = Inet::kIPAddressType_Any;
-    error                  = ChipMdnsPublishService(&service);
+    service.mProtocol = MdnsServiceProtocol::kMdnsProtocolTcp;
+    service.mPort     = CHIP_PORT;
+    TextEntry crmpRetryIntervalEntries[2];
+    uint32_t crmpRetryIntervalIdle   = CHIP_DEVICE_CONFIG_CRMP_RETRY_INTERVAL_IDLE;
+    uint32_t crmpRetryIntervalActive = CHIP_DEVICE_CONFIG_CRMP_RETRY_INTERVAL_ACTIVE;
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (chip::DeviceLayer::ConnectivityMgr().GetThreadDeviceType() ==
+        chip::DeviceLayer::ConnectivityManager::kThreadDeviceType_SleepyEndDevice)
+    {
+        uint32_t sedPollPeriod;
+        ReturnErrorOnFailure(chip::DeviceLayer::ThreadStackMgr().GetPollPeriod(sedPollPeriod));
+        // Increment default CRMP retry intervals by SED poll period to be on the safe side
+        // and avoid unnecessary retransmissions.
+        crmpRetryIntervalIdle += sedPollPeriod;
+        crmpRetryIntervalActive += sedPollPeriod;
+    }
+#endif
+    crmpRetryIntervalIdle       = Encoding::BigEndian::HostSwap32(crmpRetryIntervalIdle);
+    crmpRetryIntervalActive     = Encoding::BigEndian::HostSwap32(crmpRetryIntervalActive);
+    crmpRetryIntervalEntries[0] = { "CRI", reinterpret_cast<const uint8_t *>(&crmpRetryIntervalIdle),
+                                    sizeof(crmpRetryIntervalIdle) };
+    crmpRetryIntervalEntries[1] = { "CRA", reinterpret_cast<const uint8_t *>(&crmpRetryIntervalActive),
+                                    sizeof(crmpRetryIntervalActive) };
+
+    service.mTextEntries   = crmpRetryIntervalEntries;
+    service.mTextEntrySize = 2;
```

Comment:
Is there a macro in CHIP like `ARRAY_SIZE(crmpRetryIntervalEntries)`?

Reply:
I don't see any, there is ARRAY_LENGTH, but it has .cpp local scope. I refactored it a little bit, to increment entries counter and not hardcode size.